services/shhext/envelopes.go

```diff
@@ -10,15 +12,46 @@ import (
 	whisper "github.com/status-im/whisper/whisperv6"
 )
 
+func messageID(message whisper.NewMessage) common.Hash {
```

Comment:
Let’s make it a separate type, otherwise maps look confusing. 

`type messageID common.Hash`

Reply:
not sure if it worth it. i need to cast it in several place to common.Hash, and copy String() method. and i am it only for messageToEnvelope

i will add a comment which key is used for each map

Reply:
okay